src/coreclr/jit/flowgraph.cpp

```diff
@@ -2559,6 +2559,55 @@ PhaseStatus Compiler::fgAddInternal()
     return madeChanges ? PhaseStatus::MODIFIED_EVERYTHING : PhaseStatus::MODIFIED_NOTHING;
 }
 
+#ifdef SWIFT_SUPPORT
+//------------------------------------------------------------------------
+// fgAddSwiftErrorReturns: If this method uses Swift error handling,
+// transform all GT_RETURN nodes into GT_SWIFT_ERROR_RET nodes
+// to handle returning the error value alongside the normal return value.
+//
+// Returns:
+//   Suitable phase status.
+//
+PhaseStatus Compiler::fgAddSwiftErrorReturns()
+{
+    if (lvaSwiftErrorArg == BAD_VAR_NUM)
+    {
+        // No Swift error handling in this method
+        return PhaseStatus::MODIFIED_NOTHING;
+    }
+
+    assert(lvaSwiftErrorLocal != BAD_VAR_NUM);
+    assert(info.compCallConv == CorInfoCallConvExtension::Swift);
+
+    for (BasicBlock* block : Blocks())
+    {
+        if (block->KindIs(BBJ_RETURN))
+        {
+            GenTree* const ret = block->lastNode();
+            assert(ret->OperIs(GT_RETURN));
+            ret->SetOperRaw(GT_SWIFT_ERROR_RET);
+            ret->AsOp()->gtOp2 = ret->AsOp()->gtOp1;
+
+            // If this is the merged return block, use the merged return error local as the error operand.
+            // Else, load the error value from the SwiftError pseudolocal (this will probably get promoted, anyway).
+            if (block == genReturnBB)
+            {
+                assert(genReturnErrorLocal == BAD_VAR_NUM);
+                genReturnErrorLocal = lvaGrabTemp(true DEBUGARG("Single return block SwiftError value"));
+                lvaGetDesc(genReturnErrorLocal)->lvType = genActualType(TYP_I_IMPL);
+                ret->AsOp()->gtOp1                      = gtNewLclvNode(genReturnErrorLocal, TYP_I_IMPL);
+            }
+            else
+            {
+                ret->AsOp()->gtOp1 = gtNewLclFldNode(lvaSwiftErrorLocal, TYP_I_IMPL, 0);
+            }
+        }
+    }
```

Comment:
I still think it would be nice to do the replacement of `lvaSwiftErrorArg` with `lvaSwiftErrorLocal` here too.

Reply:
That was my plan, though I was hitting asserts around `fgNodeThreading` being invalid when trying to use any of the GenTree iterators -- `LocalsGenTreeList` seems like it would be the most efficient one for finding the `SwiftError*` arg uses. Is there a clean way of doing the IR traversal without breaking any of these iterators' invariants?

Reply:
`optReplaceScalarUsesWithConst` looks close to what you want to do I think.

Reply:
> optReplaceScalarUsesWithConst looks close to what you want to do I think.

Thanks! `GenTreeVisitor` works.